types/src/execution_result.rs

```diff
@@ -114,11 +96,11 @@ static EXECUTION_RESULT: Lazy<ExecutionResult> = Lazy::new(|| {
 #[derive(Clone, Eq, PartialEq, Serialize, Deserialize, Debug)]
 #[cfg_attr(feature = "std", derive(JsonSchema))]
 #[serde(deny_unknown_fields)]
-pub enum ExecutionResult {
+pub enum JsonExecutionResult {
```

Comment:
Why can't it still be called `ExecutionResult`? Just because it _can_ be mapped to JSON in the presentation layer doesn't really mean it _is only_ a JSON wrapper. I would keep the old name, current one confuses me. When I see a type/method that starts with `JSON*` I immediately think _"Oh, that is for the API where we use JSON"_.